src/engraving/libmscore/edit.cpp

```diff
@@ -2808,6 +2808,120 @@ void Score::deleteMeasures(MeasureBase* mbStart, MeasureBase* mbEnd, bool preser
         }
     }
 
+    if (!preserveTies) {
+        // Reconnect or remove slurs that intersect with these deleted measures
+        Fraction sTick = mbStart->tick();
+        Fraction eTick = mbEnd->tick();
+        for (auto overlappingSlur : spannerMap().findOverlapping(sTick.ticks(), eTick.ticks())) {
+            Spanner* sp = overlappingSlur.value;
+            if (!sp->isSlur()) {
+                continue;
+            }
+            ChordRest* cr1 = sp->startCR();
+            ChordRest* cr2 = sp->endCR();
+            if (!cr1 || !cr2) {
+                // this is an invalid slur
+                continue;
+            }
+            MeasureBase* m = mbStart;
+            bool adjust1 = false;
+            bool adjust2 = false;
+            do {
+                adjust1 = adjust1 ? true : cr1->measure() == m;
+                adjust2 = adjust2 ? true : cr2->measure() == m;
+                if (m == mbEnd) {
+                    break;
+                } else {
+                    m = m->next();
+                }
+            } while (m && !(adjust1 && adjust2));
+            if (adjust1 && adjust2) {
+                // if both endpoints of this slur are inside this deleted range, remove the slur.
+                undoRemoveElement(sp);
+                continue;
+            } else if (adjust1) {
+                // endpoint 1 is in deleted range. move endpoint forward to first non-deleted CR after range
+                Measure* mNext = mbEnd->next() && mbEnd->next()->isMeasure() ? toMeasure(mbEnd->next()) : nullptr;
+                if (!mNext) {
+                    undoRemoveElement(sp);
+                    continue;
+                }
+                ChordRest* firstAvailableCr = nullptr;
+                // search for last cr in this voice
+                Segment* seg = mNext->first(SegmentType::ChordRest);
+                while (seg) {
+                    firstAvailableCr = seg->cr(sp->track());
+                    if (!firstAvailableCr || !firstAvailableCr->isChord()) {
+                        seg = seg->next(SegmentType::ChordRest);
+                        continue;
+                    }
+                    auto gracesBefore = toChord(firstAvailableCr)->graceNotesBefore();
+                    if (firstAvailableCr->tick() > sp->endCR()->tick()) {
```

Comment:
Please call these sp->endCR() (and the other things that are not changes between iterations) outside the loop